File: ui-api/src/main/java/io/tipsters/uiapi/client/williamhill/OddsProvider.java
package io.tipsters.uiapi.client.williamhill;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

import io.tipsters.uiapi.dto.CompetitionMatches;

public interface OddsProvider {
  List<CompetitionMatches> odds(Set<String> competitionNames, LocalDateTime matchStart, LocalDateTime matchEnd);

  Set<String> providesCompetitions();
}

File: common/src/main/kotlin/io/tipsters/common/oddsprovider/OddsProvider.kt
package io.tipsters.common.oddsprovider

import io.tipsters.common.data.MatchesByCompetition
import java.time.LocalDateTime

interface OddsProvider {
    fun odds(competitionNames: Set<String>, matchStart: LocalDateTime, matchEnd: LocalDateTime): List<MatchesByCompetition>

    fun providesCompetitions(): Set<String>
}
